src/oscar/apps/dashboard/offers/forms.py

```diff
@@ -188,4 +203,30 @@ def save(self, *args, **kwargs):
 
 class OfferSearchForm(forms.Form):
     name = forms.CharField(required=False, label=_("Offer name"))
-    is_active = forms.BooleanField(required=False, label=_("Is active?"))
+    is_active = forms.NullBooleanField(required=False, label=_("Is active?"))
+    offer_type = forms.ChoiceField(required=False, label=_("Offer type"), choices=())
```

Comment:
Hi please just use a function to determine the available offer_types, that way you might not need to override the constructor. Choices can be a callable as described in the documentation: https://docs.djangoproject.com/en/3.1/ref/forms/fields/#django.forms.ChoiceField.choices